services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/IntentionalRerouteV2Spec.groovy

```diff
@@ -0,0 +1,268 @@
+package org.openkilda.functionaltests.spec.flows
+
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.functionaltests.helpers.PathHelper
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.info.event.PathNode
+import org.openkilda.messaging.payload.flow.FlowState
+import org.openkilda.testing.model.topology.TopologyDefinition.Switch
+import org.openkilda.testing.service.northbound.NorthboundServiceV2
+import org.openkilda.testing.service.traffexam.TraffExamService
+import org.openkilda.testing.tools.FlowTrafficExamBuilder
+import org.springframework.beans.factory.annotation.Autowired
+import spock.lang.Ignore
+import spock.lang.Narrative
+
+import javax.inject.Provider
+
+import static org.junit.Assume.assumeTrue
+import static org.openkilda.testing.Constants.DEFAULT_COST
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+@Narrative("Verify that on-demand reroute operations are performed accurately.")
+class IntentionalRerouteV2Spec extends BaseSpecification {
+
+    @Autowired
+    Provider<TraffExamService> traffExamProvider
+
+    @Autowired
+    NorthboundServiceV2 northboundV2
+
+    @Ignore("Flow status check fails due to the timeout defined in the test. Must be reconsidered.")
```

Comment:
The original version of this test is updated, I believe the new version should not fail